readme.md

```diff
@@ -7,199 +7,290 @@
+## Example of standalone modal
 
-If you need more customization use `PlainModal`. This component provides only modal frame with overlay and close button.
+#### Modal Component
+```tsx
+import { type ModalComponentProps } from '@locmod/modal2'
+import { SomePlainModalMarkUpYouHave } from 'components/ui'
 
-```typescript jsx
-import React from 'react'
-import { openModal, modalVisibility } from 'modal'
 
-import { PlainModal, ModalHeadline } from 'components/feedback'
-import type { ModalProps } from 'components/feedback'
+const SpecificModal: React.FC<ModalComponentProps> = (props) => {
+  const { closeModal } = props
+  
+  const handleClick = () => {
+    // "true" will trigger onClose handler if it will be passed to props
+    closeModal(true)
+  }
 
-
-export type CustomModalProps = {
-  username: string
+  return (
+    <SomePlainModalMarkUpYouHave closeModal={closeModal}>
+      <h3>Success!</h3>
+      <p>Your order has been placed</p>
+      <button onClick={handleClick}>
+        Ok
+      </button>
+    </SomePlainModalMarkUpYouHave>
+  )
 }
 
-const CustomModal: React.FunctionComponent<CustomModalProps & ModalProps> = ({ userName, closeModal }) => (
-  <PlainModal closeModal={closeModal}>
-    <Icon className={s.icon} name="happyBird" />
-    <ModalHeadline title="Custom title" />
-    <span>...smth else</span>
-  </PlainModal>
-)
+// you still should do it to use openModal with this name in TS
+declare global {
+  interface ModalsRegistry extends ExtendModalsRegistry<{ SpecificModal: typeof SpecificModal }> {}
+}
 
-export const openCustomModal = (props: CustomModalProps) => openModal('customModal', props)
+// should be wrapped by standaloneModal
+export default standaloneModal('SpecificModal', SpecificModal)
+```
 
+#### App
+```tsx
+import Content from 'components/Content/Content'
 
-export default modalVisibility('customModal', CustomModal)
+const App = () => (
+  <>
+    <Content />
+  </>
+)
 ```
 
-```typescript jsx
-import React, { useCallback } from 'react'
+#### Use standalone modal in Content
 
-import CustomModal, { openCustomModal } from 'compositions/modals/CustomModal/CustomModal'
+```tsx
+import { openModal } from '@locmod/modal'
+import SpecificModal from 'components/SpecificModal/SpecificModal'
 
 
-const App = () => {
-  const handleButtonClick = useCallback(() => {
-    openCustomModal({
-      userName: 'John Doe',
+const Content = () => {
+  const handleButtonClick = () => {
+    // open modal with props
+    openModal('SpecificModal', {
+      onClose: () => {
+        console.log('SpecificModal closed')
+      },
     })
-  }, [])
+  }
 
   return (
     <>
-      <button onClick={handleButtonClick}>Open modal</button>
-      <CustomModal />
+      <button type="button" onClick={handleButtonClick}>
+        Click me
+      </button>
+      {/* standalone modal should be rendered in place of usage */}
+      <SpecificModal />
```

Comment:
The examples look too long, do we want to move them to codesandbox?

Reply:
great idea
https://github.com/Local-Modules/modal/pull/1/commits/fb56bc43948815bd177e09ee84659842de44d946